modules/dnn/perf/perf_gemm.cpp

```diff
@@ -0,0 +1,148 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "perf_precomp.hpp"
+#include <opencv2/dnn/shape_utils.hpp>
+
+namespace opencv_test {
+
+struct GemmParam_t {
+    int shapeIn0[2]; // Mat A
+    int shapeIn1[2]; // Mat B
+    double declared_flops;
+};
+
+// MatA * MatB = MatC
+static const GemmParam_t testGemmConfigs[] = {
+        {{ 128 , 128 }, { 128 , 128 },  6291456.},
+        {{ 128 , 128 }, { 128 , 1024 },  50331648.},
+        {{ 128 , 128 }, { 128 , 4096 },  201326592.},
+        {{ 128 , 1024 }, { 1024 , 128 },  50331648.},
+        {{ 128 , 1024 }, { 1024 , 1024 },  402653184.},
+        {{ 128 , 1024 }, { 1024 , 4096 },  1610612736.},
+        {{ 128 , 4096 }, { 4096 , 128 },  201326592.},
+        {{ 128 , 4096 }, { 4096 , 1024 },  1610612736.},
+        {{ 128 , 4096 }, { 4096 , 4096 },  6442450944.},
+        {{ 1024 , 128 }, { 128 , 128 },  50331648.},
+        {{ 1024 , 128 }, { 128 , 1024 },  402653184.},
+        {{ 1024 , 128 }, { 128 , 4096 },  1610612736.},
+        {{ 1024 , 1024 }, { 1024 , 128 },  402653184.},
+        {{ 1024 , 1024 }, { 1024 , 1024 },  3221225472.},
+        {{ 1024 , 1024 }, { 1024 , 4096 },  12884901888.},
+        {{ 1024 , 4096 }, { 4096 , 128 },  1610612736.},
+        {{ 1024 , 4096 }, { 4096 , 1024 },  12884901888.},
+        {{ 1024 , 4096 }, { 4096 , 4096 },  51539607552.},
+        {{ 4096 , 128 }, { 128 , 128 },  201326592.},
+        {{ 4096 , 128 }, { 128 , 1024 },  1610612736.},
+        {{ 4096 , 128 }, { 128 , 4096 },  6442450944.},
+        {{ 4096 , 1024 }, { 1024 , 128 },  1610612736.},
+        {{ 4096 , 1024 }, { 1024 , 1024 },  12884901888.},
+        {{ 4096 , 1024 }, { 1024 , 4096 },  51539607552.},
+        {{ 4096 , 4096 }, { 4096 , 128 },  6442450944.},
+        {{ 4096 , 4096 }, { 4096 , 1024 },  51539607552.},
+        {{ 4096 , 4096 }, { 4096 , 4096 },  206158430208.},
+};
+
+struct GemmParamID
+{
+    enum {
+        GEMM_0 = 0,
+        GEMM_LAST = sizeof(testGemmConfigs) / sizeof(testGemmConfigs[0])
+    };
+
+    int val_;
+    GemmParamID(int val = 0) : val_(val) {}
+    operator int() const { return val_; }
+    static ::testing::internal::ParamGenerator<GemmParamID> all()
+    {
+        enum { NUM = (int)GEMM_LAST };
+        GemmParamID v_[NUM]; for (int i = 0; i < NUM; ++i) { v_[i] = GemmParamID(i); } // reduce generated code size
+        return ::testing::ValuesIn(v_, v_ + NUM);
+    }
+};
```

Comment:
Why do we need this struct? Can we just go through all the test cases by default?

Reply:
Please see :https://github.com/opencv/opencv/blob/4.x/modules/dnn/perf/perf_convolution1d.cpp#L29-L44